examples/systemcalls/systemcalls.c

```diff
@@ -93,6 +167,88 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
+
```

Comment:
@Suraj-Ajjampur think about how you could avoid duplicating logic between these two functions.

Reply:
Added Function to perform the fork execv and also check if redirection is required.

```
bool execute_command(char * command[], int redirect_fd)
{
    pid_t child_pid = fork();

    if (child_pid == -1) {
        // Fork failed
        syslog(LOG_ERR, "Fork failed");
        return false;
    }

    if (child_pid == 0) {
        // This is the child process

        // Redirect stdout if necessary
        if (redirect_fd != -1) {
            if (dup2(redirect_fd, 1) == -1) {
                syslog(LOG_ERR, "Failed to redirect stdout");
                exit(EXIT_FAILURE);
            }
        }

        // Execute the command
        if (execv(command[0], command) == -1) {
            syslog(LOG_ERR, "execv failed");
            exit(EXIT_FAILURE);
        }
    } else {
        // This is the parent process
        int status;

        // Wait for the child process to finish
        wait(&status);

        if (WIFEXITED(status)) {
            return (WEXITSTATUS(status) == 0);
        } else {
            return false;
        }
    }

    return true;  // Should never reach here
}
```